### PR TITLE
[RFC] switch to functions for libc, atomic support, etc checks 

### PR DESCRIPTION
**Pull request by ericonr**
- I just threw the functions in `shutils/common.sh`, they can certainly go elsewhere. They haven't undergone extensive testing at all.
- The packages with additional changes in the second commit (`xbps_target_libc`) should get their own commits.

Possible additions: `xbps_target_wordsize`, `xbps_host_target_wordsize_differ` (or something to that effect).

### CODE REVIEW

**Comment by fosslinux** [comment]
Why?

I don't understand the rationale behind this at all, why make this a wrapper that IMHO for libc at least makes it look less clean?? `$XBPS_TARGET_LIBC = "musl"` is much easier to read than `if xbps_target_libc musl`....

File: common/xbps-src/shutils/common.sh [hunk 1]
```diff
@@ -657,3 +657,14 @@ setup_pkg() {
         . $XBPS_BUILDHELPERDIR/${f}.sh
     done
 }
+
+xbps_target_libc() {
+    local libc="$1"
+    [ "$XBPS_TARGET_LIBC" = "$libc" ]
+    return $?
```
[the remark this answers is not in the document]
Comment (ericonr) [reply common/xbps-src/shutils/common.sh:664]: Mostly wanted to give a name to the parameters that are used.

**Comment by ericonr** [comment]
@fosslinux 

> $XBPS_TARGET_LIBC = "musl" is much easier to read than if xbps_target_libc musl....

I kinda think the opposite :P . @the-maldridge and @q66 had expressed interest in an idea similar to this, which is why I went ahead on trying to implement it.

**Comment by q66** [comment]
this is haphazard and in its current form completely pointless

**Comment by fosslinux** [comment]
@q66 @the-maldridge What idea did you have interest in?

**Comment by Duncaen** [comment]
Adding new functions also makes parsing template files even more complicated, now you need a new set of functions in addition to the environment variables.

**Comment by ericonr** [comment]
Closing this pending on a better design, then.

**Pull request closed without merging**